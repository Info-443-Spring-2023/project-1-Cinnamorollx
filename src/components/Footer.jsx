import React from 'react'

export default function Footer() {
    return (
        <footer>
            <div className="container-footer" data-testid="container-footer">
                <p className='p-4'>OurFamily is a University of Washington course project.</p>
            </div>
        </footer>
    );
}
