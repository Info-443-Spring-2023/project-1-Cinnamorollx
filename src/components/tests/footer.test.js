import { toHaveStyle } from '@testing-library/jest-dom'
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Footer from '../Footer'

describe("Testing footer rendering", () => {
    test("render footer text correctly", () => {
        render(<Footer/>);
        expect(screen.getByText("OurFamily is a University of Washington course project.")).toBeInTheDocument();
    })

    describe("Testing footer styling", () => {
        test("correct background color", () => {
            render(<Footer/>);
            const footer = screen.getByTestId("container-footer");
            expect(footer).toHaveStyle(`background-color: "#ece4da"`);
        })
        test('text changes style upon hover', () => {
            render(<Footer/>);
            const footer = screen.getByTestId("container-footer");
            userEvent.hover(footer);
            expect(footer).not.toHaveStyle({height: "7vh"});
        });
    })
})
