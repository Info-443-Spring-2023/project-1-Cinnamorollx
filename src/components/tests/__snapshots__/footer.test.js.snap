// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer rendering test match snapshot 1`] = `undefined`;
